azext_edge/tests/edge/checks/int/helpers.py

```diff
@@ -109,19 +108,47 @@ def assert_eval_core_service_runtime(
             ]
             pod_conditions = kubectl_pods[pod]["status"].get("conditions", {})
 
-            for condition_type, condition_key in conditions_to_evaluate:
-                condition_status = assert_pod_condition(pod_conditions, pod_evals, condition_type, condition_key)
-                if condition_status == "error":
-                    expected_status = "error"
-                    if namespace_status != "error":
-                        namespace_status = expected_status
-                    if overall_status != "error":
-                        overall_status = expected_status
+            known_conditions = [condition[0] for condition in conditions_to_evaluate]
+            unknown_conditions = [
+                condition["type"] for condition in pod_conditions if condition["type"] not in known_conditions
+            ]
+            # if all known conditions in pod_conditions are "True", set is_known_success to True
+            is_known_success = False
+            if all(condition_statuses(pod_conditions, known_conditions)):
+                is_known_success = True
+
+            if not is_known_success:
+                expected_status = "error"
+
+            if is_known_success and unknown_conditions:
+                # if all known conditions are True, but there are unknown conditions, set status to warning
+                expected_status = "warning"
+
+            assert_pod_conditions(pod_conditions, phase_conditions_eval, expected_status)
+
+            if namespace_status != "error":
+                namespace_status = expected_status
+            if overall_status != "error":
+                overall_status = expected_status
+
+            assert phase_conditions_eval["status"] == expected_status
 
         assert runtime_resource[namespace]["status"] == namespace_status
     assert post_deployment["evalCoreServiceRuntime"]["status"] == overall_status
 
 
+def assert_pod_conditions(pod_conditions, phase_conditions_eval, expected_status):
+    if phase_conditions_eval:
+        for condition in pod_conditions:
+            condition_type = condition.get("type")
```

Comment:
Weren't you saying "type" is guaranteed to be there in previous comment ?

Reply:
yeah, will update to keep consistent